pkg/quay/quay.go

```diff
@@ -464,5 +464,18 @@ func (c *QuayClient) DeleteTag(organization, repository, tag string) (bool, erro
 	if res.StatusCode == 404 {
 		return false, nil
 	}
-	return false, fmt.Errorf("error deleting tag")
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return false, fmt.Errorf("failed to read response body, error: %s", err)
+	}
+	data := &QuayError{}
+	err = json.Unmarshal(body, data)
+	if err != nil {
+		return false, fmt.Errorf("failed to unmarshal body, error: %s", err)
+	}
+	if data.Error != "" {
```

Comment:
it is super strange that quay sometimes returns content in `error_message` and sometimes in `error`. good spot thought

would it be worth to add this condition to other methods as well? for example I noticed that in case of an error in `DeleteRepository` method the error is present in the `error` field as well, not in `error_message`, as we're returning [here](https://github.com/redhat-appstudio/image-controller/pull/22/commits/c31e8607e77f16033914a353493028014eadf9b0#diff-d4a36e5f4e14c5636dfed4031b618cc2a45afc79934eabb0336a1adde773fd1bL137)
<img width="772" alt="Screenshot 2023-04-21 at 13 12 03" src="https://user-images.githubusercontent.com/4881144/233622251-283bd527-f7fb-435c-a395-914bf1fa1871.png">

Reply:
Unless there is a nicer way to handle both possibilities, I can do it, sure

Reply:
probably worth to check whether there's even some API endpoint that we're calling in these methods where we can expect `error_message` field populated in case of an error. maybe we don't need `error_message` field at all

Reply:
if it's simple enough to fix it (which in this case I think it is), I'd do it in this PR, just a separate commit